external-crates/move/move-compiler/src/diagnostics/codes.rs

```diff
@@ -42,6 +42,16 @@ pub trait DiagnosticCode: Copy {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+/// Represents a single annotation for a diagnostic filter
+pub enum WarningFilter {
+    All,
+    Category(Category),
+    Code(Category, /* code */ u8),
+}
```

Comment:
a comment, even a verbose one to explain what those are would be great for me. I still do not understand what code is.
Having said all of that if having this "as_is" is much more in line with the rest of the compiler than ignore me